lab-davide/app/entry.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+// webpack is copying the below into the build directory//
+require('!!file?name=[name].[ext]!./html/index.html');
+require('./scss/base.scss'); //leave ext on sass file//
+
+//npm module//
+const angular = require('angular');
+const cowsay = require('cowsay-browser');
+
+//defining a module. using an empty array because we are not using other angular modules that are dependent   on this module//
+var app = angular.module('cowsayApp', []);
+
+//$scope is a injectible predefined piece of functionality that we can add to a controller//
+app.controller('cowsayController', [ '$scope', function($scope) {
```

Comment:
Controllers are constructor functions, so their names must be capitalized. 

If I were to write this, it would look like this:


```javascript
const cowsay = require('cowsay');

// IIFEs are pretty standard angular practice. They aren't strictly needed w/ webpack, but they help to separate backend code, like require statements, from frontend code. 
(function(){
  // It's best practice to never create a variable called app to store your module. 
  angular.module('cowsayApp', [])
    .controller('CowsayController', [
       '$scope',
       CowsayController,
    ]);

  function CowsayController($scope) {
    // PROPERTIES 
    $scope.cowsayText         = 'blah';
    $scope.animalThatsTalking = 'default';

    // METHODS
    $scope.say    = say;
    $scope.done  = done;

    // DEFINITIONS
    
    // This runs the text in the input field through cowsay to generate preview text
    function say() {
      // stuff
    }

    // This populates the result with the result of the cowsay function
    function done() {
      // stuff 
    }

  }

});
```